src/main/java/de/tum/in/www1/artemis/domain/User.java

```diff
@@ -93,6 +93,10 @@
     @ElementCollection
     private List<String> groups = new ArrayList<>();
 
+    @JsonIgnore
+    @Column(name = "guided_tour_settings")
+    private String guidedTourSettings = null;
```

Comment:
I just thought about this again and I think this would be a bad solution, because of the following reasons:
1) The json string might be quite long and the keys are saved for each user, wasting a lot of unnecessary space
2) The attribute will be loaded every time for each user in each scenario

I think a separate table, that is lazy loaded just in case this information is needed, would be better.
The table would have the userId as foreign key and would have each tutorial as column with a boolean value. Then each user entry could be stored very efficiently in the database, e.g. 1,0,1,0,1 and it would not need around 100 characters of string values per user.

I think this would be modeled with an entity object, e.g. GuidedTourSettings, with booleans. Each user would then have a OneToOne association to such an object and we would use the following annotation for this association.

@OneToOne(fetch = FetchType.LAZY)

Before we deploy this to production, I would ask you to change the entity model. Thanks!

Reply:
We do not even need a Boolean. If it exist it is true, if it does not exist it is false. The table would map user it and a key which we can be an enum.

Reply:
Ok, even better: Maybe look at this to store it as a HasMap using JPA: http://www.java2s.com/Tutorials/Java/JPA/0320__JPA_ElementCollection_MapKeyEntity.htm

Reply:
I keep coming back to this one for some reason. I think as most of the logic here is actually part of the client, I think it would be nice to design the `guidedTourSettings` as a plain key-value store. This way future tours can be implemented on the client-side without any modification to the server side, not even an enum update would be needed. Maybe think of this also when revisiting the decision made here.

Reply:
@krusche @jpbernius 
Thank you for your feedback.

I can move the guidedTourSettings to a new table and link it to the User table with the user_id as foreign key and only fetch them if necessary.
But I don't know if having an own column for each setting would be good to maintain since we would have to adjust the database for every change. If we keep the settings in a JSON string, however, then we wouldn't have to make server-side adjustments for new guided tours. What do you think about this?

Reply:
No, the table would be a compound primary key with user if and setting key as keys. And an possible value.
Check the Database die the User to group Mapping.